### PR TITLE
optim: Math.hypot replace Math.sqrt(Math.pow(x,2))

### PR DESCRIPTION
**Pull request by jack-shanghai**
~~it seems targetTo function have some problems, according to my analysis, we should invert the 'out' matrix.~~ Besides, I replace Math.sqrt(Math.pow()+...) with Math.hypot() which is more elegant.

### CODE REVIEW

File: src/mat2.js [hunk 1]
```diff
@@ -291,7 +291,7 @@ export function str(a) {
  * @returns {Number} Frobenius norm
  */
 export function frob(a) {
-  return(Math.sqrt(Math.pow(a[0], 2) + Math.pow(a[1], 2) + Math.pow(a[2], 2) + Math.pow(a[3], 2)))
+  return(Math.hypot(...a))
```
Comment (stefnotch) [inline src/mat2.js]: I would prefer to not use the spread operator, as I'm not entirely sure about it's performance.

Comment (jack-shanghai) [reply src/mat2.js]: that ok, how about Math.hypot to replace Math.sqrt(Math.pow)

**Comment by stefnotch** [comment]
Regarding targetTo, it's supposed to be this way. For more information, check out https://github.com/toji/gl-matrix/issues/225

However, the Math.hypot optimization seems to be pretty interesting. 😄

File: src/mat2.js (hunk 1 above)
Comment (stefnotch) [reply src/mat2.js]: Yes, of course. That's pretty cool!

File: src/mat4.js [hunk 2]
```diff
@@ -1513,7 +1512,7 @@ export function targetTo(out, eye, target, up) {
   out[13] = eyey;
   out[14] = eyez;
   out[15] = 1;
-  return out;
+  return invert(out, out);
```
Comment (stefnotch) [inline src/mat4.js]: As stated in #225 , this does not need to be inverted. If you want the inverse, you should use `lookAt`

Comment (jack-shanghai) [reply src/mat4.js]: ok, due to my poor English, I still can't understand #225 said, it seems lookAt function has achieved the goal which I thought target should would do, Thank you!

Comment (stefnotch) [reply src/mat4.js]: @jack-shanghai Yes, that is correct. 

Would you mind changing this pull request from `fix: targetTo..` to `optim: Math.hypot..`?

Comment (jack-shanghai) [reply src/mat4.js]: sorry, I forget it, it's done

**Comment by stefnotch** [comment]
👍 Thank you for the great pull request! 😄 I'll merge it ASAP.

**Comment by doodlewind** [comment]
`Math.hypot` is interesting indeed 👍 A concern is that is gl-matrix supposed to be compatible with IE? `hypot ` is landed in ES2015 so it may not work on IE11.

**Comment by stefnotch** [comment]
Yes, that is a valid concern indeed. There are a few options here:
- Drop IE support: Sub-optimal
- Reject the PR: Also sub-optimal, as gl-matrix is supposed to be super fast
- Transpile `Math.hypot` to `Math.pow(Math.sqrt(...`
- Ship a `Math.hypot` polyfill with gl-matrix

**Comment by doodlewind** [comment]
How about detecting the browser's `Math.hypot` support and use our current implementation as fallback? In this way we don't have to include the heavy official polyfill libs, neither do we need extra transpile process.

**Comment by stefnotch** [comment]
> heavy official polyfill libs

I didn't suggest adding a fully blown polyfill library. We only have to polyfill, `Math.hypot`, which is [rather easy and lightweight](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/hypot#Polyfill). 😄 
However, the polyfill option does have a disadvantage: gl-matrix is going to be slower on IE. But I suppose that's an acceptable downside.

**Comment by jack-shanghai** [comment]
How can you be so patient, it's my pleasure to do my bit!

**Pull request merged**